plugins/com.google.cloud.tools.eclipse.login.test/src/com/google/cloud/tools/eclipse/login/ui/AccountsPanelTest.java

```diff
@@ -106,13 +122,22 @@ public void testAccountsArea_threeAccounts() {
     AccountsPanel panel = new AccountsPanel(null, loginService);
     panel.createDialogArea(shell);
 
-    assertEquals(3, panel.accountLabels.size());
-    String text1 = panel.accountLabels.get(0).getText();
-    String text2 = panel.accountLabels.get(1).getText();
-    String text3 = panel.accountLabels.get(2).getText();
-    assertTrue((text1 + text2 + text3).contains(account1.getEmail()));
-    assertTrue((text1 + text2 + text3).contains(account2.getEmail()));
-    assertTrue((text1 + text2 + text3).contains(account3.getEmail()));
+    assertEquals(3, panel.emailLabels.size());
+    verifyLabelsContains(panel.emailLabels, "alice@example.com");
+    verifyLabelsContains(panel.emailLabels, "bob@example.com");
+    verifyLabelsContains(panel.emailLabels, "charlie@example.com");
+    verifyLabelsContains(panel.nameLabels, "Alice");
+    verifyLabelsContains(panel.nameLabels, "");
+    verifyLabelsContains(panel.nameLabels, "Charlie");
+  }
+
+  private void verifyLabelsContains(List<Label> labels, String text) {
```

Comment:
That feels like a bug in itself. Should the order of these things be deterministic?

Even if it isn't there are better ways to test for this, rather than looping repeatedly.